src/devices/Board/I2cBusExtensions.cs

```diff
@@ -44,5 +44,16 @@ public static List<int> PerformBusScan(this I2cBus bus, int lowest = 0x3, int hi
 
             return ret;
         }
+
+        /// <summary>
+        /// Create I2cBus Object type from I2cDevice
+        /// </summary>
+        /// <param name="i2cDevice">The I2cDevice for which I2cBus object needs to be created</param>
+        /// <returns></returns>
+        public static I2cBus CreateBusFromI2CDevice(this I2cDevice i2cDevice)
+        {
+            return I2cBus.Create(i2cDevice.ConnectionSettings.BusId);
+        }
+
```

Comment:
Unfortunately, that is not going to work, because it's not generic. 

The I2cDevice you have in hand here does not necessarily point to a device on the primary I2c Controller, it could be pointing to a device attached to an USB-To-I2c controller such as the FT4222. In this case, you would get something wrong or an exception here, as the system doesn't necessarily have a default controller at all. 

I think, to work around this, you would need to have your binding operate on the bus, not an individual device. Which logically doesn't seem to be that wrong to me.

Reply:
So Should I create an I2CBus for the device and from that create I2CDevices for the multiplexer?

Reply:
I would like to know what approach will be better for this?

Reply:
I need to think about it for a bit and will make a suggestion tonight. But yes, it will probably be something like that.

Reply:
What about this:

Change the constructor of `Tca9548A` to:
```csharp
 public Tca9548A(I2cBus masterBus, int multiplexerAddress, bool shouldDisposeBus = true)
{
    _masterBus = masterBus ?? throw new ArgumentNullException(nameof("masterBus"));
    _multiplexerDevice = masterBus.CreateDevice(multiplexerAddress);
}
```

So you specify that this multiplexer operates on the given bus and has the given address. 

You would instantiate it like so:
```csharp
var bus = I2cBus.Create(1);
var tca = new Tca9458A(bus, Tca9548A.DefaultI2cAddress);
// ...
```
or actually better yet (as more generic:)

```csharp
var board = new RaspberryPiBoard();
var bus = board.CreateOrGetI2cBus(1);
var tca = new Tca9458A(bus, Tca9548A.DefaultI2cAddress);
// ...
```

and then, in your `Tca9458A` class, you add a method like:

```csharp
public I2cBus CreateBusForChannel(int channel)
{
     return new Tca9548Bus(this, channel);
}
```
with Tca9548Bus being an internal class that would automatically remember that it was in fact one of the slave busses of the TCA.

Similarly, you had

```csharp
public I2cDevice CreateDevice(I2cConnectionSettings settings)
{
    // Verify settings.Channel is a valid sub-channel ID.
     return new Tca9458Bus(this, settings.Channel).CreateDevice(settings.DeviceAddress);
}
```
(this method is actually just a shortcut on what the bus would offer as well). The actual instance of I2cDevice returned would also be an internal implementation remembering that it was a part of the TCA. 

This all looks a bit complicated at first, but I think it will help keep things universally usable, even with special top-level controllers. And it has the nice effect that you could forward the obtained I2cDevice instance to any binding without having to care that it's attached to a multiplexer, as the channel switching would happen behind the scenes. 

I'm not currently suggesting to remove the existing `SelectChannel` method, as there might be use cases where you want to still do that manually (e.g. if you want to talk to multiple devices in parallel). 

What do you think?

Reply:
Hi, 
I was also thinking the best way would be to pass an I2cBus object type in the constructor. 

But I am confused about creating different I2Cbus for all the channels because all the devices connected to the TCA9548A mux will be on the same bus on which TCA9548A multiplexer is.

Reply:
Yes, the trick would be to create artificial bus numbers. So `CreateBusForChannel(1)` would return bus number 0x101 (or maybe `own_device_I'd << 8 + channel` so it would even work with multiple tca's).

Reply:
Hi, I have updated the object creation method for  TCA, but I was still not able to implement the artificial bus creation for different channel